test/gurka/test_tagged_names.cc

```diff
@@ -0,0 +1,134 @@
+         {"CD", {{"highway", "motorway"}, {"tunnel:name", "Fort McHenry Tunnel"}}},
+         {"DE", {{"highway", "motorway"}, {"tunnel:name", "Fort McHenry Tunnel"}}},
+         {"EF", {{"highway", "motorway"}}},
+         {"BG", {{"highway", "motorway"}}},
+         {"GH", {{"highway", "motorway"}}}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize_metres);
+    map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_tagged_names", build_config);
+  }
+};
+gurka::map TaggedNames::map = {};
+Api api;
+rapidjson::Document d;
+
+/*************************************************************/
+
+TEST_F(TaggedNames, Tunnel) {
+  auto result = gurka::route(map, "A", "F", "auto");
+
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  auto leg = result.trip().routes(0).legs(0);
+  EXPECT_EQ(leg.node(0).edge().has_tunnel(), false);
+  EXPECT_TRUE(leg.node(1).edge().tagged_name().Get(0).type() == TaggedName_Type_kTunnel);
+  EXPECT_TRUE(leg.node(1).edge().tagged_name().Get(0).value() == "Fort McHenry Tunnel");
+  EXPECT_TRUE(leg.node(2).edge().tagged_name().Get(0).type() == TaggedName_Type_kTunnel);
+  EXPECT_TRUE(leg.node(2).edge().tagged_name().Get(0).value() == "Fort McHenry Tunnel");
+  EXPECT_TRUE(leg.node(3).edge().tagged_name().Get(0).type() == TaggedName_Type_kTunnel);
+  EXPECT_TRUE(leg.node(3).edge().tagged_name().Get(0).value() == "Fort McHenry Tunnel");
+  EXPECT_EQ(leg.node(4).edge().has_tunnel(), false);
+  EXPECT_EQ(leg.node(5).edge().has_tunnel(), false);
+}
+
+TEST_F(TaggedNames, NoTunnel) {
+  auto result = gurka::route(map, "A", "H", "auto");
+
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  auto leg = result.trip().routes(0).legs(0);
+  EXPECT_EQ(leg.node(0).edge().has_tunnel(), false);
+  EXPECT_EQ(leg.node(1).edge().has_tunnel(), false);
+  EXPECT_EQ(leg.node(2).edge().has_tunnel(), false);
+}
+
+TEST_F(TaggedNames, test_taking_tunnel) {
+  std::string locations = R"({"lon":)" + std::to_string(map.nodes["A"].lng()) + R"(,"lat":)" +
+                          std::to_string(map.nodes["A"].lat()) + R"(},{"lon":)" +
+                          std::to_string(map.nodes["F"].lng()) + R"(,"lat":)" +
+                          std::to_string(map.nodes["F"].lat()) + "}";
+
+  auto reader = std::make_shared<baldr::GraphReader>(map.config.get_child("mjolnir"));
+  valhalla::tyr::actor_t actor(map.config, *reader, true);
+  auto json = actor.route(R"({"costing":"auto","format":"osrm","locations":[)" + locations + R"(]})",
+                          {}, &api);
+
+  // get the osrm json
+  d.Parse(json);
+  EXPECT_FALSE(d.HasParseError());
+
+  for (const auto& route : d["routes"].GetArray()) {
+    for (const auto& leg : route["legs"].GetArray()) {
+      for (const auto& step : leg["steps"].GetArray()) {
+        for (const auto& intersection : step["intersections"].GetArray()) {
+          if (intersection.HasMember("tunnel_name")) {
```

Comment:
This test will never fail if the `tunnel_name` property is missing. We can make it robust to that by explicitly stating which intersections we expect to have the tunnel_name property, e.g.

```EXPECT_EQ(intersections[0]["tunnel_name"].GetString(), "Fort McHenry Tunnel")```